docs/two-phase-commit-transactions.md

```diff
@@ -0,0 +1,174 @@
+```Java
+TwoPhaseCommitTransaction tx = manager.join("<transaction ID>")
+```
+
+You need to specify the transaction ID associated with the transaction that the coordinator has started.
+
+
+#### Resume the transaction
+
+You can get the transaction object (the `TwoPhaseCommitTransaction` instance) that you have already joined with `TwoPhaseCommitTransactionManager.resume()`:
+```Java
+TwoPhaseCommitTransaction tx = manager.resume("<transaction ID>")
+```
+
+`TwoPhaseCommitTransactionManager` manages the transaction objects that you have joined, and you can get it with the transaction ID.
+
+### CRUD operations for the transaction
+
+The way to execute CRUD operations in Two-phase Commit Transactions is the same as the transaction API.
+`TwoPhaseCommitTransacton` has `get()`/`put()`/`delete()`/`mutate()` to execute CRUD operations.
+
+This is an example code for CRUD operations in Two-phase Commit Transactions:
+```java
+TwoPhaseCommitTransaction tx = ...
+
+// Retrieve the current balances for ids
+Get fromGet = new Get(new Key(ID, fromId));
+Get toGet = new Get(new Key(ID, toId));
+Optional<Result> fromResult = tx.get(fromGet);
+Optional<Result> toResult = tx.get(toGet);
+
+// Calculate the balances (it assumes that both accounts exist)
+int newFromBalance = fromResult.get().getValue(BALANCE).get().getAsInt() - amount;
+int newToBalance = toResult.get().getValue(BALANCE).get().getAsInt() + amount;
+
+// Update the balances
+Put fromPut = new Put(new Key(ID, fromId)).withValue(BALANCE, newFromBalance);
+Put toPut = new Put(new Key(ID, toId)).withValue(BALANCE, newToBalance);
+tx.put(fromPut);
+tx.put(toPut);
+```
+
+### Prepare/Commit/Rollback the transaction
+
+After finishing CRUD operations, you need to commit the transaction.
+Like a well-known two-phase commit protocol, there are two phases: prepare and commit phases.
+you need to prepare the transaction in all the coordinator/participant processes first and then commit them as follows:
+```Java
+TwoPhaseCommitTransaction tx = ...
+
+try {
+  // Execute CRUD operations in the coordinator/participant processes
+  ...
+
+  // Prepare phase: Prepare the transaction in all the coordinator/participant processes
+  tx.prepare();
+  ...
+
+  // Commit phase: Commit the transaction in all the coordinator/participant processes
+  tx.commit()
+  ...
+} catch (TransactionException e) {
+  // When an error happans, you need to rollback the transaction in all the coordinator/participant processes
+  tx.rollback();
+  ...
+}
+```
+
+You need to call `rollback()` in all the coordinator/participant processes when an error happens.
+
+You can call `prepare()` parallelly in the coordinator/participant processes.
+Note that you need to call `commit()` in the coordinator process first.
+After that, you can call `commit()` parallelly in the participant processes.
+Similarly, you need to call `rollback()` in the coordinator process first, and then you can call it parallelly in the participant processes.
```

Comment:
```suggestion
```
Moved up since I think it should be noted earlier with the description of `validate()`.